diff --git a/internal/core/model/repeater.rs b/internal/core/model/repeater.rs
index 381a29a..47884de 100644
--- a/internal/core/model/repeater.rs
+++ b/internal/core/model/repeater.rs
@@ -331,7 +331,7 @@ fn update_visible_instances(
         // Jumping more than 1.5 screens: random seek.
         ops.splice(0, ops.len(), 0);
         state.offset = ((-content_y_value / element_height).floor() as usize).min(row_count - 1);
-        (state.offset, 0 as Coord)
+        (state.offset, content_y_value + element_height * state.offset as Coord)
     } else if content_y_value < state.previous_content_y {
         // Scrolled down: find the new offset by walking existing instances.
         let mut it_y = first_item_y + content_y_value;
@@ -370,6 +370,9 @@ fn update_visible_instances(
         if prepend_count > 0 {
             state.offset = new_offset;
         }
+        if new_offset == 0 && new_offset_y > 0 as Coord && content_y_value <= 0 as Coord {
+            new_offset_y = 0 as Coord;
+        }
         debug_assert!(new_offset >= state.offset && new_offset <= state.offset + ops.len());
 
         // Layout items until we fill the view, starting with already-instantiated ones.
