diff --git a/anda/apps/opensnitch/opensnitch.spec b/anda/apps/opensnitch/opensnitch.spec
index 653df93..0311c0d 100644
--- a/anda/apps/opensnitch/opensnitch.spec
+++ b/anda/apps/opensnitch/opensnitch.spec
@@ -23 +23 @@ Name:           opensnitch
-Release:        2%{?dist}
+Release:        3%{?dist}
@@ -39,0 +40,8 @@ BuildRequires:  qt6-linguist
+Requires:       python3dist(grpcio) >= 1.10.1
+Requires:       python3dist(packaging)
+Requires:       python3dist(protobuf)
+Requires:       python3dist(pyqt6) >= 6.4
+Requires:       python3dist(python-slugify) >= 7.0.0
+Recommends:     python3-inotify
+Recommends:     python3-notify2
+Recommends:     python3dist(requests)
@@ -55,0 +64 @@ sed -i 's/@pip3 /@python3 -m pip /' ui/Makefile
+sed -i 's|/usr/local/bin/opensnitchd|%{_bindir}/opensnitchd|' daemon/data/init/opensnitchd.service
@@ -82,0 +92,4 @@ install -Dm644 daemon/data/init/opensnitchd.service %{buildroot}%{_unitdir}/open
+install -Dm644 daemon/data/default-config.json daemon/data/system-fw.json daemon/data/network_aliases.json \
+  -t %{buildroot}%{_sysconfdir}/opensnitchd
+install -Dm600 daemon/data/rules/*.json -t %{buildroot}%{_sysconfdir}/opensnitchd/rules
+install -Dm600 daemon/data/tasks/tasks.json -t %{buildroot}%{_sysconfdir}/opensnitchd/tasks
@@ -87,0 +101,2 @@ popd
+install -Dm644 ui/resources/opensnitch_ui.desktop \
+  -t %{buildroot}%{_sysconfdir}/xdg/autostart
@@ -106,0 +122,9 @@ rm -rf %buildroot%python3_sitelib%_usr
+%dir %{_sysconfdir}/opensnitchd
+%config(noreplace) %{_sysconfdir}/opensnitchd/default-config.json
+%config(noreplace) %{_sysconfdir}/opensnitchd/network_aliases.json
+%config(noreplace) %{_sysconfdir}/opensnitchd/system-fw.json
+%dir %{_sysconfdir}/opensnitchd/rules
+%config(noreplace) %{_sysconfdir}/opensnitchd/rules/*.json
+%dir %{_sysconfdir}/opensnitchd/tasks
+%config(noreplace) %{_sysconfdir}/opensnitchd/tasks/tasks.json
+%{_sysconfdir}/xdg/autostart/opensnitch_ui.desktop
