diff --git a/packages/wrangler/src/__tests__/secret.test.ts b/packages/wrangler/src/__tests__/secret.test.ts
index 312e3b4..fef8c10 100644
--- a/packages/wrangler/src/__tests__/secret.test.ts
+++ b/packages/wrangler/src/__tests__/secret.test.ts
@@ -275,6 +275,15 @@ describe("wrangler secret", () => {
 				expect(std.err).toMatchInlineSnapshot(`""`);
 			});
 
+			it("should refuse an empty piped secret value", async () => {
+				mockStdIn.send("");
+				await expect(
+					runWrangler("secret put the-key --name script-name")
+				).rejects.toThrowError(
+					"Need to pass in a value when creating a secret."
+				);
+			});
+
 			it("should create a secret, from piped input", async ({ expect }) => {
 				mockPutRequest(expect, { name: "the-key", text: "the-secret" });
 				// Pipe the secret in as three chunks to test that we reconstitute it correctly.
diff --git a/packages/wrangler/src/secret/index.ts b/packages/wrangler/src/secret/index.ts
index 8ef1875..6d37b02 100644
--- a/packages/wrangler/src/secret/index.ts
+++ b/packages/wrangler/src/secret/index.ts
@@ -138,6 +138,13 @@ export const secretPutCommand = createCommand({
 				: await readFromStdin()
 		);
 
+		if (!secretValue) {
+			throw new UserError(
+				"Need to pass in a value when creating a secret.",
+				{ telemetryMessage: "secret put missing value" }
+			);
+		}
+
 		logger.log(`🌀 Creating the secret for the Worker "${scriptName}"`);
 
 		async function submitSecret() {
