// subtle/astro.config.mjs
import { defineConfig } from 'astro/config';
import mdx from '@astrojs/mdx';
import sitemap from '@astrojs/sitemap';
import { browserslistToTargets, composeVisitors } from 'lightningcss';
import { browserslist } from 'browserslist';
import { resolveToEsbuildTarget } from 'esbuild-plugin-browserslist';
import apply from '@monochromatic.dev/lightningcss-plugin-apply';
import customUnits from '@monochromatic.dev/lightningcss-plugin-custom-units';
import propertyLookup from '@monochromatic.dev/lightningcss-plugin-property-lookup';
import size from '@monochromatic.dev/lightningcss-plugin-size';

// https://astro.build/config
export default defineConfig({
  site: 'https://monochromatic.dev/subtle',
  integrations: [mdx(), sitemap()],
  redirects: {
    '/posts/[...slug]': '/post/[...slug]',
    '/blog/[...slug]': '/post/[...slug]',
    '/blogs/[...slug]': '/post/[...slug]',
    '/article/[...slug]': '/post/[...slug]',
    '/articles/[...slug]': '/post/[...slug]',
  },
  scopedStyleStrategy: 'where',
  vite: {
    css: {
      devSourcemap: true,
      transformer: 'lightningcss',
      lightningcss: {
        targets: browserslistToTargets(browserslist()),
        cssModules: { pattern: '[local]' },
        analyzeDependencies: true,
        visitor: composeVisitors([apply, customUnits, propertyLookup, size]),
        drafts: { customMedia: true },
      },
    },
    esbuild: {
      minifyIdentifiers: false,
    },
    build: {
      target: resolveToEsbuildTarget(browserslist()),
      cssMinify: 'lightningcss',
    },
    worker: {
      format: 'es',
    },
  },
  build: {
    inlineStylesheets: 'always',
  },
  prefetch: {
    defaultStrategy: 'viewport',
  },
  markdown: {
    shikiConfig: {
      experimentalThemes: {
        light: 'github-light',
        dark: 'github-dark',
      },
      wrap: true,
    },
  },
  i18n: {
    defaultLocale: 'en',
    locales: ['en', 'zh'],
    fallback: {},
  },
  experimental: {
    optimizeHoistedScript: true,
    contentCollectionCache: true,
    clientPrerender: true,
  },
});
